libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -417,6 +417,37 @@ def ListDiskByLabels(self,
     return self._ListByLabel(
         labels_filter, disk_service_object, filter_union)
 
+  def ListReservedExternalIps(self, zone: str) -> List[str]:
+    """Lists all static external IP addresses that are reserved (i.e. not in
+    use) and can be used in a particular zone.
+
+    The method first converts the zone to a region,
+    and then queries the GCE addresses resource.
+
+    Args:
+      zone (str): The zone in which the returned IPs would be available.
+
+    Returns:
+      List[str]: The list of available IPs in the specified zone.
+    """
+    # Convert zone to region
+    zone_parts = zone.split('-')
+    if len(zone_parts) != 3:
+      raise ValueError("Invalid zone: {}.")
+    region = '-'.join(zone_parts[:-1])
+    # Request list of addresses
+    addresses_client = self.GceApi().addresses()
+    request = addresses_client.list(project=self.project_id, region=region)
+    response = request.execute()
```

Comment:
You can use ExecuteRequest directly and it will handle retries and paging, it returns a list of responses (respnse pages) other than a single response, so you would also need to loop on it  https://github.com/google/cloud-forensics-utils/blob/a849b99e780726969fd3fecf8ccab1d4ddcc5ebd/libcloudforensics/providers/gcp/internal/common.py#L265

Reply:
Done, thanks for pointing it out. The API call is indeed a "paged" response, and ExecuteRequest handles that case 👍